### PR TITLE
[Brave News]: Better offline handling & UX tweaks

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/35153
Resolves https://github.com/brave/brave-browser/issues/35155
Resolves https://github.com/brave/brave-browser/issues/35154
Resolves https://github.com/brave/brave-browser/issues/35157

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

**Note:** When testing this the company VPN should be off, as Brave can't detect the network state and requests hang (this happens for `fetch` requests from JavaScript too).

1. Start Brave with no internet
2. An error message for "You appear to be offline" should show up
3. Connect
4. Refresh
5. Feed should load
6. Go offline
7. Change feed (should work fine)
8. Refresh
9. Offline message should show
10. (see individual issues for ux tweaks issues)

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/brave_news/browser/resources/Feed.tsx [hunk 1]
```diff
@@ -145,18 +151,12 @@ export default function Component({ feed, hasSubscriptions }: Props) {
...
-
   return <FeedContainer className={NEWS_FEED_CLASS}>
     {feed
-      ? noArticles
-        ? hasSubscriptions ? <NoArticles /> : <NoFeeds />
-        : <>
-          {cards}
-          <CaughtUp />
-        </>
+      ? errors[feed.error!] ?? <>
```
Comment (sangwoo108) [inline components/brave_news/browser/resources/Feed.tsx:156]: I'm curious if this is a hacky way to do the same with
```
(feed.error && errors[feed.error]) ??
```
, as `errors[null]` would be undefined anyway

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: components/brave_news/browser/brave_news_controller.h [hunk 2]
```diff
@@ -160,6 +169,9 @@ class BraveNewsController : public KeyedService,
   void Prefetch();
   void MaybeInitPrefs();
 
+  base::Time last_update_;
+  bool was_offline_{false};
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply components/brave_news/browser/brave_news_controller.h:173]: Removed (it was unused :fearful: )

File: components/brave_news/browser/resources/Feed.tsx (hunk 1 above)
Comment (fallaciousreasoning) [reply components/brave_news/browser/resources/Feed.tsx:156]: Exactly - errors has an exact list of keys it allows, but I wanted to treat it as a map which may or may not have the key. Just felt it read a bit nicer, but happy to change it :smile:

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using

```

```

## Case One: https://github.com/brave/brave-browser/issues/35153 - `PENDING`
## Case Two: https://github.com/brave/brave-browser/issues/35155 - `PENDING`
## Case Three: https://github.com/brave/brave-browser/issues/35154 - `PENDING`
## Case Four: https://github.com/brave/brave-browser/issues/35157 - `PENDING`

**Pull request merged**